packages/replay-next/src/suspense/EventsCache.ts

```diff
@@ -160,3 +160,239 @@ export function getEventAsync(eventType: EventHandlerType, point: ExecutionPoint
 export function getEventIfCached(eventType: EventHandlerType, point: ExecutionPoint) {
   return getEventCache(eventType).getResultIfCached(point);
 }
+
+/**
+ * Encode a Chromium event type, based on its category and non-unique event type.
+ */
+function makeChromiumEventType(nonUniqueEventType: string, category?: string) {
+  if (category) {
+    return `${category}.${nonUniqueEventType}`;
+  }
+
+  // we were not able to match this event type to a pre-defined category
+  return nonUniqueEventType;
+}
+
+/**
+ * NOTE: The `rawEventType` is produced in MakeReplayMakeEventType in chromium's
+ *   `InspectorDOMDebuggerAgent`.
+ * @see https://github.com/replayio/chromium/blob/509fb5f3c3a587ebef668a631a53eed83acc2f85/third_party/blink/renderer/core/inspector/inspector_dom_debugger_agent.cc#L940
+ */
+function decodeChromiumEventType(rawEventType: string) {
+  const [nonUniqueEventType, eventTargetName] = rawEventType.split(",", 2);
+  return [nonUniqueEventType, eventTargetName];
+}
+
+async function getRecordingTarget() {
+  if (recordingTarget === null) {
+    recordingTarget = await ThreadFront.getRecordingTarget();
+  }
+  return recordingTarget;
+}
+
+async function countEvents(eventCountsRaw: Record<string, number>) {
+  const recordingTarget = await getRecordingTarget();
+  const targetCountEvents = CountEvents[recordingTarget];
+  if (!targetCountEvents) {
+    if (!isEmpty(eventCountsRaw)) {
+      console.error(
+        `Internal Error: recordingTarget "${recordingTarget}" has events but is missing CountEvents function`
+      );
```

Comment:
This doesn't look right. Node recordings are configured with `supportsEventTypes: true` but this new implementation would always log an error and count no events.

Reply:
Uhm... Why do Node recordings have `supportsEventTypes`?
Also what do you recommend, remove the `if`?

Reply:
https://github.com/replayio/devtools/blob/b1a5e77285d64c269e17fe3f5e81810a0baae83a/packages/protocol/thread/thread.ts#L249-L257

Reply:
Should I remove the `if`?

Reply:
If you did, it would break with your current implementation– wouldn't it? (There is no `CountEvents["node"]`.)

That seems like an oversight or a disagreement at least in my understanding of what Node was supposed to support. Maybe my understanding was just wrong though and the "capabilities" code had the wrong default on the client. Could totally be that! In which case, the thing to do would be to edit the `thread/thread.ts` code to match.

Reply:
I think, you added those flags [here](https://github.com/replayio/devtools/pull/7713/files#diff-6e85b32eceb414e3df67e264b8594beb834c30fd38ce5b8ffa774dcd08dabc9eR259) 6 months ago?

From a runtime perspective, I don't think we would expect Node events to work. The `OnEvent` function is also not being used [here](https://github.com/replayio/node/blob/1f7d292d2cc117103eb1ca4246d870b0e7a0cf75/deps/v8/src/api/api.cc#L10893).
So, I'll just go ahead and set it to `false` for Node.